presto-spi/src/test/java/com/facebook/presto/spi/type/TestTypeSignature.java

```diff
@@ -156,11 +156,10 @@ public void parseRowSignature()
                 "row(time time with time zone,array(interval day to seconds))");
 
         // preserve base name case
-        assertSignature(
+        assertRowSignature(
                 "RoW(a bigint,b varchar)",
-                "RoW",
-                ImmutableList.of("a bigint", "b varchar"),
-                "row(a bigint,b varchar)");
+                rowSignature(namedParameter("a", false, signature("bigint")), namedParameter("b", false, varchar())));
+        assertEquals(parseTypeSignature("RoW(a bigint,b varchar)").getBase(), "RoW");
```

Comment:
I don't think this is necessary: `assertEquals(parseTypeSignature("RoW(a bigint,b varchar)").getBase(), "RoW")`. The previous line has already validated that the base of the base name is preserved.